docs/详细设计方案.md

```
@@ -457,13 +463,13 @@ async function run(state: AppState) {
 
 > **[2026-08-31 架构决策更新] 任务 4.4 Coordinator Ledger + 角色切换交接定稿。** 采用 MagenticOne 双循环的结构但拒绝其 LLM 裁决：L1 `core/domain/coordination-ledger.ts` 定义并严格校验 JSON schema；L2 `core/orchestration/progress-ledger.ts` 以纯函数 `buildCoordinationLedger(state, observation)` 生成外层 `task={confirmedFacts,hypotheses,plan[]}` 与内层 `progress={isRequestSatisfied,isInLoop,isProgressBeingMade,nextSpeaker,instructionOrQuestion}`。字段遵 R10 使用 camelCase；`plan[]` 为 `{id,revision,role,instruction,status,dependsOn}`。台账不新增 AppState/Mutation 字段，而作为 `Message{fromRole:'COORDINATOR',type:'chat',payload.kind:'coordination_ledger'}` 经 `appendMutation('messages',...)` 持久化：`chat` 类型确保不污染 4.3 只扫描 announce/feedback/escalation 的控制事件恢复；读取端只接受完整 shape，畸形嵌套 plan/facts 直接忽略。每次 `decide()` 先产生原 Route/Mutation，再由统一 `attachCoordinationArtifacts` 基于 mutation 后的结构化 State 追加 Ledger，故 `nextSpeaker` 与实际 Route 同源。
 >
-> 停滞语义：显式 `tests_failed` / `repeated_test_failures` / `review_changes_requested` / `reviewer_architecture_issue` 回环，或 progressMarker（phase/complexity/requirements/architecture/subtask 状态/test/review/humanGate；明确排除 iterationCount）未变化，均令 `isInLoop=true`、`isProgressBeingMade=false` 并累计 stall；有结构化进展则清零；`maxStalls=3` 时 revision+1、重建 facts/plan、stall 清零。机器只能置 `completionCandidate`；Phase 4 的 `isRequestSatisfied.answer` 恒为 false，完成候选理由为 `awaiting_leader_confirmation` 且 `authority='leader'`，普通完成路径不提前进入 humanGate；Phase 8 接入 Leader resolve 与 D4 持久 suspend/真 Fork resume 后才可置真。
+> 停滞语义：显式 `tests_failed` / `repeated_test_failures` / `review_changes_requested` / `reviewer_architecture_issue` 回环，或 progressMarker（phase/complexity/requirements/architecture/subtask 状态/test/review/humanGate；明确排除 iterationCount）未变化，均令 `isInLoop=true`、`isProgressBeingMade=false` 并累计 stall；有结构化进展则清零；`maxStalls=3` 时 revision+1、重建 facts/plan、stall 清零。机器只能置 `completionCandidate`；Phase 4–7 的 `isRequestSatisfied.answer` 恒为 false，完成候选理由为 `awaiting_leader_confirmation` 且 `authority='leader'`。自 Phase 8/D16 起，当前 REVIEWER `approved` 先进入绑定该 verdict 的 `completion_confirmation` D4 gate；只有经规范 Leader completion resolution 交叉验证后才可置真。`request_changes` 保持 false 并进入返工，旧 review 的裁决不得越轮复用。
 >
 > 同一统一接缝解决 DEF-012：当 `state.nextRole` 存在且新 worker role 不同时，Coordinator 通用追加 `HandoffPacket`（同角色重派/初次派发不生成），`done` 记录阶段交接，`keyDecisions` 引用 State 中 leader 决策 id，`openIssues/fileRefs` 从结构化测试失败提取；所有包仍过 reducer 的 shape/引用完整性门禁。不是只硬编码旧五条 flip，4.2/4.3 新增 REVIEWER/ARCHITECT 路径也自动覆盖。真实 Phase 2 Harness/MCP/LocalTempSandbox 链在无 seed 情况下验证 PM→ARCHITECT→CODER→TESTER→REVIEWER 四次交接和每个安全点 Ledger 事件。
 
```

markdownlint-cli2 (0.23.2)

[warning] Blank line inside blockquote

(MD028, no-blanks-blockquote)